hypergan/trainers/alpha_trainer.py

```diff
@@ -0,0 +1,60 @@
+import tensorflow as tf
+import numpy as np
+import hyperchamber as hc
+import inspect
+
+from hypergan.trainers.base_trainer import BaseTrainer
+
+TINY = 1e-12
+
+class AlphaTrainer(BaseTrainer):
+    def __init__(self, gan, config, losses=[], var_lists=[]):
+        BaseTrainer.__init__(self, gan, config)
+        self.losses = losses
+        self.var_lists = var_lists
+
+    def _create(self):
+        gan = self.gan
+        config = self.config
+        losses = self.losses
+        g_lr = config.g_learn_rate
+        d_lr = config.d_learn_rate
+
+        optimizers = []
+        self.d_lr = tf.Variable(d_lr, dtype=tf.float32)
+        self.g_lr = tf.Variable(g_lr, dtype=tf.float32)
+        for i, _ in enumerate(losses):
+            loss = losses[i]
+            var_list = self.var_lists[i]
+
+            if i ==0 or i == 1:
+                optimizer = self.build_optimizer(config, 'g_', config.g_trainer, self.g_lr, var_list, loss)
+            else:
+                optimizer = self.build_optimizer(config, 'd_', config.d_trainer, self.d_lr, var_list, loss)
+            optimizers.append(optimizer) #TODO prefx
+
```

Comment:
This is the second place where it would be nice to output a set subtraction of tf.global_variables and all the variables that have a gradient defined in any of the optimizers.  This would allow us to give warnings if a stray variable existed anywhere in the graph.